### PR TITLE
Cleanup brave shields util

### PR DESCRIPTION
**Pull request by simonhong**
Resolves https://github.com/brave/brave-browser/issues/10078

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: components/brave_shields/browser/brave_shields_util.cc [hunk 1]
```diff
@@ -382,48 +349,48 @@ ControlType GetFingerprintingControlType(Profile* profile, const GURL& url) {
...
 
   if (!primary_pattern.IsValid())
     return;
 
-  HostContentSettingsMapFactory::GetForProfile(profile)
-      ->SetContentSettingCustomScope(
-          primary_pattern, ContentSettingsPattern::Wildcard(),
-          ContentSettingsType::PLUGINS, kHTTPUpgradableResources,
-          CONTENT_SETTING_DEFAULT);
+  map ->SetContentSettingCustomScope(
```
Comment (iefremov) [inline components/brave_shields/browser/brave_shields_util.cc:379]: extra space

File: components/brave_shields/browser/brave_shields_util.h [hunk 2]
```diff
@@ -19,73 +19,91 @@ struct Referrer;
...
+
 enum ControlType { ALLOW = 0, BLOCK, BLOCK_THIRD_PARTY, DEFAULT, INVALID };
 
 ContentSettingsPattern GetPatternFromURL(const GURL& url);
 std::string ControlTypeToString(ControlType type);
 ControlType ControlTypeFromString(const std::string& string);
 
-void SetBraveShieldsEnabled(Profile* profile, bool enable, const GURL& url);
+void SetBraveShieldsEnabled(HostContentSettingsMap* map,
+                            PrefService* local_state,
```
Comment (iefremov) [inline components/brave_shields/browser/brave_shields_util.h:35]: maybe we can make this a last parameter with a default value? Just to make the interface less ugly

File: components/brave_shields/browser/cookie_pref_service_browsertest.cc [hunk 3]
```diff
@@ -4,6 +4,7 @@
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #include "brave/components/brave_shields/browser/brave_shields_util.h"
+#include "chrome/browser/content_settings/host_content_settings_map_factory.h"
```
Comment (iefremov) [inline components/brave_shields/browser/cookie_pref_service_browsertest.cc:7]: Shouldn't we move the browsertest to `brave/browser`?

File: browser/net/url_context.cc [hunk 4]
```diff
@@ -98,13 +99,19 @@ void BraveRequestInfo::FillCTX(const network::ResourceRequest& request,
...
   Profile* profile = Profile::FromBrowserContext(browser_context);
   ctx->allow_brave_shields =
-      brave_shields::GetBraveShieldsEnabled(profile, ctx->tab_origin);
-  ctx->allow_ads = brave_shields::GetAdControlType(profile, ctx->tab_origin) ==
-                   brave_shields::ControlType::ALLOW;
+      brave_shields::GetBraveShieldsEnabled(
+          HostContentSettingsMapFactory::GetForProfile(profile),
+          ctx->tab_origin);
+  ctx->allow_ads = brave_shields::GetAdControlType(
+      HostContentSettingsMapFactory::GetForProfile(profile),
```
Comment (iefremov) [inline browser/net/url_context.cc:106]: Can we save this to a local var and pass to all of the calls?

File: components/brave_shields/browser/cookie_pref_service_browsertest.cc (hunk 3 above)
Comment (simonhong) [reply components/brave_shields/browser/cookie_pref_service_browsertest.cc:7]: Moved to brave/browser/brave_shields/...

**Comment by simonhong** [comment]
Merged because only test-install was failed.

**Pull request merged**